easybuild/framework/easyblock.py

```diff
@@ -2434,6 +2437,9 @@ def run_step(self, step, step_methods):
         """
         self.log.info("Starting %s step", step)
         self.update_config_template_run_step()
+
+        run_hook(step, self.hooks, pre_step_hook=True, args=[self])
```

Comment:
You mean, test that `self` is being passed as an argument?

That's already done in , see `test_toy_build_hooks` in `toy_build.py` where the `post_install_hook` there which uses `self.name` and `self.version`.

Reply:
I mean that `args` seems like a kind of wildcard that I find trouble understanding. It is passed directly to the hook in all cases except the start and end hooks, where `run_hook` is executed without any of the optional arguments. And then the hook itself might do something with it, but being it either `[]` or `[self]` I don't find it very clear what is the purpose of it. I guess that now, with a slightly better understanding of what you are doing here, my comment is simply "can we rename `args` to something more descriptive?". With a different name I suppose I would understand it from the beginning.

Reply:
For now, only the step hooks get passed an argument (i.e. `self`, the `EasyBlock` instance that gives access to all information).

This may change in the future though, if a reason pops up to pass additional arguments.

`args` is commonly used in Python code, see for example https://pythontips.com/2013/08/04/args-and-kwargs-in-python-explained/.

The idea is that the `args` list is passed down with `*args` to "unpack" it when the hook is called.

Hooks should be implemented like this, to be future-proof (i.e. keep working when we may pass down additional (even named) arguments):

```
def pre_install_hook(self, *args, **kwargs):
    # do something with self

def start_hook(*args, **kwargs):
    # do something (no (named) arguments passed, for now)
```

I'll clarify this in the yet-to-be-written documentation.